Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp

```diff
@@ -1035,6 +1032,27 @@ static ErrorOr<NonnullOwnPtr<BitBuffer>> generic_refinement_region_decoding_proc
         return result;
     };
 
+    // Figure 13 – 10-pixel refinement template
+    constexpr auto compute_context_1 = [](BitBuffer const& reference, int reference_x, int reference_y, BitBuffer const& buffer, int x, int y) -> u16 {
+        u16 result = 0;
+
+        for (int dy = -1; dy <= 1; ++dy) {
+            for (int dx = -1; dx <= 1; ++dx) {
+                if ((dy == -1 && (dx == -1 || dx == 1)) || (dy == 1 && dx == -1))
+                    continue;
+                result = (result << 1) | (u16)get_pixel(reference, reference_x + dx, reference_y + dy);
+            }
+        }
+
+        for (int i = 0; i < 3; ++i)
+            result = (result << 1) | (u16)get_pixel(buffer, x - 1 + i, y - 1);
+        result = (result << 1) | (u16)get_pixel(buffer, x - 1, y);
+
+        return result;
+    };
+
+    u16 (*compute_context)(BitBuffer const&, int, int, BitBuffer const&, int, int) = inputs.gr_template == 0 ? compute_context_0 : compute_context_1;
```

Comment:
Would be nice to not need to fully declare the function pointer param list - does this work instead?
```c++
auto compute_context = inputs.gr_template == 0 ? compute_context_0 : compute_context_1;
```
clangd seems to resolve it to a function pointer at least:
![Screenshot_20240325_121609](https://github.com/SerenityOS/serenity/assets/5600524/e3940ab3-2462-4faa-b56b-a6550f3fc657)

Reply:
Oh that's much nicer, thanks!

(`generic_region_decoding_procedure` a bit further down does something similar with 4 templates, and there it's long enough that I can't use a ternary, so I need to write out the type there. I tried an IIFE there, but return type deduction in a lambda gives me the lambda type, so I don't immediately see a way to prevent typing out the type over there.)

Reply:
Played around with this a bit just out of curiosity, this should work, though still a bit of a mouthful:
```c++
EquivalentFunctionType<decltype(compute_context_0)>* compute_context = nullptr;
```

Or as an IIFE:
```c++
auto compute_context = ([&]() -> EquivalentFunctionType<decltype(compute_context_0)>* {
    if (inputs.gb_template == 0)
        return compute_context_0;
    if (inputs.gb_template == 1)
        return compute_context_1;
    if (inputs.gb_template == 2)
        return compute_context_2;
    VERIFY(inputs.gb_template == 3);
    return compute_context_3;
})();
```